### PR TITLE
feat: Render icon for empty text-enabled plugins when editing

### LINKED ISSUE #140: Embedded plugin not visible in TextPlugin TipTap editor
A "text enabled" plugin does not show inside TextPlugin editor.
The plugin shows on the structure board but there is no way to
click on it from the tip-tap editor

STEPS TO REPRODUCE
- set the `text_enabled` attribute to `True` on the `CMSPluginBase` instance
- Create an instance of `TextPlugin`
- Click on the `CMS Plugin` dropdown inside the tip-tap editor
- Add the "text-enabled" plugin and save.
- On edit, it shows in the structure board, but there is no way to access it from
   the tip-tap editor

### PR DESCRIPTION
**Pull request by fsbraun**
## Summary by Sourcery

Render visible placeholders for zero-width CMS plugins in the Tiptap editor so users can see and interact with empty text-enabled plugins.

New Features:
- Add automatic detection of zero-width CMS plugins and render a visible icon placeholder so they remain selectable and editable in the editor.

Enhancements:
- Style empty CMS plugins and their placeholder icons for consistent alignment and visual appearance in the editor UI.

Tests:
- Add JavaScript tests for the CMS plugin behavior around rendering placeholders for empty plugins.

fixes #140

### CODE REVIEW

**Review by mrbazzan — COMMENTED** [review]
Hi @fsbraun

Consider this example in a text-editor:
```text editor
<p><cms-plugin></cms-plugin>Hello[space-0]</p>

<p>[space-1]</p>
```

If I insert a cms-plugin in `space-0`, TipTap treats as an 
inline node and the initial dimension is close to 5x18 (inline layout of cms-plugin inside the paragraph).
> The browser creates a line box for that paragraph. Even an empty inline element can end up with about `5x18`px

On Safari, this breaks the `dom.offsetWidth === 0 || dom.offsetHeight` condition causing the icon not to show
For `space-1`, there is no text, so dimension is close to 0x18


Instead of using offsetWidth and offsetHeight, what about using the new custom `hasVisibleContent` function?

File: private/js/tiptap_plugins/cms.plugin.js [hunk 1]
```diff
@@ -210,6 +210,30 @@ const cmsPluginNodes = {
             });
             // store the getPos function in the node to be able to edit the node from the menu bar later
             node.getPos = getPos;
+            // Detect descendant visiblilty after layout. Insert the plugin's icon
+            // (or a generic puzzle icon) as a real DOM child so the user can
+            // still see, select, and edit the plugin.
+	    function hasVisibleContent(el){
+	        for (const child of el.querySelectorAll('*')){
+		    const rect = child.getBoundingClientRect();
+		    if (rect.width > 0 && rect.height > 0) return true;
```
Comment (fsbraun) [inline private/js/tiptap_plugins/cms.plugin.js]:
```suggestion
		    if (rect.width > 18 && rect.height > 18) return true;
```

**Comment by fsbraun** [comment]
@mrbazzan Thans for the fix!
